communication/src/ping.h

```diff
@@ -34,11 +42,30 @@ class Pinger
 		 * USER  SYS      NO
 		 * USER  USER     YES
 		 */
+		// TODO: It feels that this logic should have been implemented in the system layer
```

Comment:
I agree with this.  The comms layer just provides the timeout and the system layer manages the multiple stakeholders that may choose to change it.  Similarly to how we manage the LED.